# Paired T test examples - Examples of Different Formats
#### By Melissa Gonzalez and Marcy Misner 
--- 
+ Below is a [link](http://www.sthda.com/english/wiki/paired-samples-t-test-in-r#:~:text=The%20paired%20samples%20t%2Dtest,t%2Dtest%20using%20R%20software.&text=Compute%20the%20mean%20(m)%20and,standard%20deviation%20(s)%20of%20d) that has more information on when to use either a tilde or a comma in your t.test() function.

+ See the image and further examples below:

![A screenshot of t test examples to illustrate when to use tilde or commas.](../Media/t-test-examples.png)



# Examples
## Using skydive data in a paired t-test using vectors
+ First, we're going to access our dataset. We will use the dataset used in DSO101 that contains heart rates of individuals before and after skydiving.

In [8]:
skydive <- read.csv("../../DS101-Basic-Statistics/Data/skydive.csv")

+ We imported this dataset into a variable named ```skydive```
+ To look at this dataset, we need to call the dataset the function ```View(skydive)``` to view the entire dataset.

In [10]:
View(skydive)

subjGrp time heartRt
1   1      1    73.78  
2   2      1    79.60  
3   3      1    81.37  
4   4      1    85.46  
5   5      1    85.03  
6   6      1    67.81  
7   7      1    64.79  
8   8      1    84.82  
9   9      1    78.31  
10 10      1    68.13  
11 11      1    76.91  
12  1      5    91.47  
13  2      5    89.23  
14  3      5    93.32  
15  4      5    89.59  
16  5      5    77.07  
17  6      5    90.56  
18  7      5    79.37  
19  8      5    84.20  
20  9      5    95.09  
21 10      5    71.90  
22 11      5    86.18

+ This dataset has:
    + 11 different indivduals listed as a number in the column ```subjGrp```, 
    + time (before = 1, after = 5) in the column ```time```,
    + and heart rates in the column ```heartRt```
    
+ We will now put all of the 'before' values and all of the 'after' values into separate vectors.
    + Rows 1 - 11 have the before values
    + Rows 12 - 22 have the after values

In [13]:
before <- skydive$heartRt[1:11]
after <- skydive$heartRt[12:22]

+ We can now run a paired t-test using these two vectors of heart rate measurements to compare.
+ Because they are in vector format, we will use commas to separate the variables in our t.test function.
+ the basic syntax is ```t.test(vector1, vector2, paired = TRUE)```

In [25]:
skyDiveVectortTest <- t.test(before, after, paired = TRUE)

+ To see the results of our ```skyDiveVectortTest```, we will now call it by running just the name.

In [20]:
skyDiveVectortTest


	Paired t-test

data:  before and after
t = -3.4463, df = 10, p-value = 0.006265
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -15.263312  -3.276688
sample estimates:
mean of the differences 
                  -9.27 


+ The p-value from this t-test is 0.006265. Because this is less than our alpha of 0.05, we will reject the null hypothesis that there is no difference in heart rate between before and after skydiving. We will instead accept the alternative hypothesis that there IS a different between the before and after heartrates.

+ Next we will look at the same data, but run the t-test using the data in a dataframe format.
---
## Using skydive data in a paired t-test using a dataframe format
+ We can also run a t-test if we keep the data in the format of the dataframe.
+ To do this, we would list the column with the heart rate values to compare (```heartRt```) followed by a tilde (```~```) and then list the column that lists the information relating to groups. In this situation, this information is in the column named ```time```.
+ We will then need to list the name of the dataframe.
+ The basic syntax is ```t.test(values ~ grouping, data = dataFrameName, paired = TRUE)```

In [22]:
skyDiveDataFrametTest <- t.test(heartRt ~ time, data = skydive, paired = TRUE)

In [23]:
skyDiveDataFrametTest


	Paired t-test

data:  heartRt by time
t = -3.4463, df = 10, p-value = 0.006265
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -15.263312  -3.276688
sample estimates:
mean of the differences 
                  -9.27 


+ Notice how the resulting p-value is exactly the same. The only difference is the format of how we fed the same information to run this paired t-test.